src/main/java/uk/gov/companieshouse/confirmationstatementapi/controller/FilingController.java

```diff
@@ -1,22 +1,38 @@
 package uk.gov.companieshouse.confirmationstatementapi.controller;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
-import uk.gov.companieshouse.api.model.transaction.Transaction;
 import uk.gov.companieshouse.api.model.filinggenerator.FilingApi;
+import uk.gov.companieshouse.confirmationstatementapi.exception.SubmissionNotFoundException;
+import uk.gov.companieshouse.confirmationstatementapi.service.FilingService;
 
 @RestController
 @RequestMapping("/private/transactions/{transaction_id}/confirmation-statement/{confirmation_statement_id}/filings")
 public class FilingController {
 
-    @GetMapping
-    public ResponseEntity<FilingApi> getFiling(@RequestAttribute("transaction") Transaction transaction) {
-        var filing = new FilingApi();
-        filing.setDescription(transaction.getDescription());
+    private static final Logger LOGGER = LoggerFactory.getLogger(FilingController.class);
+
+    @Autowired
+    private FilingService filingService;
 
-        return ResponseEntity.ok(filing);
+    @GetMapping
+    public ResponseEntity<FilingApi> getFiling(@PathVariable("confirmation_statement_id") String confirmationStatementId) {
+        FilingApi filing = null;
```

Comment:
The filing is only referenced inside the try block so you can change line 28 to 
FilingApi filing = filingService.generateConfirmationFiling(confirmationStatementId);
and remove line 26